conjureup/controllers/bootstrap/common.py

```diff
@@ -0,0 +1,90 @@
+import os
+
+from conjureup import controllers, juju, utils
+from conjureup.app_config import app
+from conjureup.telemetry import track_event
+
+
+class BootstrapController:
+    def __init__(self, msg_cb):
+        self.msg_cb = msg_cb
+
+    def render(self):
+        app.loop.create_task(self.do_bootstrap(app.current_credential))
+        controllers.use('deploy').render()
+
+    async def do_bootstrap(self, creds):
+        if app.current_controller is None:
+            app.current_controller = "conjure-up-{}-{}".format(
+                app.current_cloud,
+                utils.gen_hash())
+
+        if app.current_model is None:
+            app.current_model = utils.gen_model()
+
+        if not app.is_jaas:
+            await self.pre_bootstrap()
+            app.log.info('Bootstrapping Juju controller.')
+            self.msg_cb('Bootstrapping Juju controller.')
+            track_event("Juju Bootstrap", "Started", "")
+            cloud_with_region = app.current_cloud
+            if app.current_region:
+                cloud_with_region = '/'.join([app.current_cloud,
+                                              app.current_region])
+            success = await juju.bootstrap(app.current_controller,
+                                           cloud_with_region,
+                                           app.current_model,
+                                           credential=creds)
+            if not success:
+                pathbase = os.path.join(
```

Comment:
Does it make sense to update these path creation to use Path instead? I think it would be nice to be able to assume that you can interact with any paths in the same fashion. This is just a question not something to do at the moment though